### PR TITLE
Misc improvements (WIP)

### PR DESCRIPTION
**Pull request by revmischa**
* Some support for documentation generation - not quite finished
* Switch from yarn -> npm ([stats](https://ashleynolan.co.uk/blog/frontend-tooling-survey-2019-results#js-packagemanagers))
* Add ability to run init-db and seed on lambda directly as functions instead of using flask CLI
* Improved "drop all tables" script for those times when DB cascades fail for init-db
* Add staging stage
* `make deploy-xx`
* Fix (sorta - mypy seems really iffy about this) the location of mypy cache files to not be inside the project
* Use serverless aurora for database by default. It can go to sleep after a predefined time, which should save us lots of money for dev environment databases. Right now serverless aurora is not available in `eu-central-1` region so we're using Ireland for now.
* Remove generated DB URL secret and construct the URL in python instead of CF
* NPlusOne

I would like to port over our new improved VPC setup so we can properly isolate the database on its own private subnet with no internet access, with a bastion EC2 host. I would like to figure out a way to do this without costing us lots of money, which is currently an issue because of the NAT gateways required when you put lambda functions in a private VPC subnet.

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -12,14 +12,16 @@ custom:
...
       DeletionProtection: true
   stackName: ${self:service}-${self:provider.stage}
-  xray: 'true'  # AWS Xray tracing of requests, queries
+  xray: 'Active'  # AWS Xray tracing of requests, queries
   db:
-    dbname: "TEMPLATE"
-    instance_name: "#{AWS::StackName}"
-    instance_class: db.t3.small
-    storage: '10'  # GB
+    dbname: "TEMPLATEx"
```
Comment (revmischa) [inline serverless.yml]: needed for testing

File: TEMPLATE/config.py [hunk 2]
```diff
@@ -15,6 +17,8 @@ class Config:
     SECRETS_NAME = os.getenv("APP_SECRETS_NAME", "TEMPLATE/dev")
     RDS_SECRETS_NAME = os.getenv("RDS_SECRETS_NAME")
 
+    DEV_DB_SCRIPTS_ENABLED = True
```
Comment (gnudeb) [inline TEMPLATE/config.py]: I feel like `False` would be a better default for this one

File: .circleci/config.yml [hunk 3]
```diff
@@ -32,16 +35,46 @@ workflows:
       - type_checks:
           requires:
             - deps
+  deployment:
```
Comment (adamszeptycki) [inline .circleci/config.yml]: 👍 for comments

File: TEMPLATE/config.py [hunk 4]
```diff
@@ -47,6 +51,9 @@ class Config:
     JWT_ACCESS_TOKEN_EXPIRES = timedelta(hours=8)
     CAN_SEED_DB = True
 
+    NPLUSONE_LOGGER = logging.getLogger("app.nplusone")
```
Comment (adamszeptycki) [inline TEMPLATE/config.py]: is this enabled by default? (I'm 99% sure that observing these extra queries is adding some overhead)

File: TEMPLATE/create_app.py [hunk 5]
```diff
@@ -34,6 +35,7 @@ def create_app(test_config=None) -> App:
     app.wsgi_app = ProxyFix(app.wsgi_app)  # type: ignore
     configure_database(app)
     api.init_app(app)  # flask-rest-api
+    NPlusOne(app)
```
Comment (adamszeptycki) [inline TEMPLATE/create_app.py]: maybe only for dev

File: TEMPLATE/config.py (hunk 4 above)
Comment (revmischa) [reply TEMPLATE/config.py]: how much overhead do you imagine?

Comment (adamszeptycki) [reply TEMPLATE/config.py]: not much, but would like us to be aware of how much extra (not needed) stuff is done. Also, we are not going to read the logs every day so it's wasted IMHO.

Comment (revmischa) [reply TEMPLATE/config.py]: IMO if warnings or errors are popping up we should be reading/doing something about them.
it's my belief that some simple accounting is vastly insignificant compared to say, network requests or queries or disk operations. you're right that we shouldn't pile up unneeded stuff. i just think the potential benefit of premature CPU optimization for accounting is outweighed by the potential for performance improvements finding n+1 queries, even or _especially_ on production. they shouldn't be ignored either, and if it turns out they are giving us false positives or things we can't do anything about we can disable it.

**Pull request merged**